src/ios/PushPlugin.m

```diff
@@ -607,4 +620,97 @@ -(void)stopBackgroundTask:(NSTimer*)timer
     }
 }
 
+
+- (void)pushRegistry:(PKPushRegistry *)registry didUpdatePushCredentials:(PKPushCredentials *)credentials forType:(NSString *)type
+{
+    if([credentials.token length] == 0) {
+        NSLog(@"VoIPPush Plugin register error - No devide token:");
+        return;
+    }
+
+    NSLog(@"VoIPPush Plugin register success");
+    const unsigned *tokenBytes = [credentials.token bytes];
+    NSString *sToken = [NSString stringWithFormat:@"%08x%08x%08x%08x%08x%08x%08x%08x",
+                        ntohl(tokenBytes[0]), ntohl(tokenBytes[1]), ntohl(tokenBytes[2]),
+                        ntohl(tokenBytes[3]), ntohl(tokenBytes[4]), ntohl(tokenBytes[5]),
+                        ntohl(tokenBytes[6]), ntohl(tokenBytes[7])];
+
+    NSLog(@"VoIPPush Plugin device token %@", sToken);
+    NSMutableDictionary* results = [NSMutableDictionary dictionaryWithCapacity:2];
+    [results setObject:sToken forKey:@"registrationId"];
+    [results setObject:@"APNS" forKey:@"registrationType"];
+
+    CDVPluginResult* pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsDictionary:results];
+    [pluginResult setKeepCallback:[NSNumber numberWithBool:YES]]; //[pluginResult setKeepCallbackAsBool:YES];
+    [self.commandDelegate sendPluginResult:pluginResult callbackId:self.callbackId];
+}
+
+- (void)pushRegistry:(PKPushRegistry *)registry didReceiveIncomingPushWithPayload:(PKPushPayload *)payload forType:(NSString *)type
+{
+    NSLog(@"Notification received");
+    NSDictionary* notificationMessage = payload.dictionaryPayload;
+    if (notificationMessage && self.callbackId != nil)
```

Comment:
So from this part of the method all the way to the end it appears to be a duplicate of the `notificationReceived` method. So instead of duplicating the code could you not write this as:

```
- (void)pushRegistry:(PKPushRegistry *)registry didReceiveIncomingPushWithPayload:(PKPushPayload *)payload forType:(NSString *)type
{
    NSLog(@"Notification received");
    self.notificationMessage = payload.dictionaryPayload;
    [self notificationReceived];
}
```

Reply:
Ok